transaction-flow.md

```diff
@@ -0,0 +1,594 @@
+# The transaction flow
+
+- [Introduction](#introduction)
+- [Creating transactions](#creating-transactions)
+  - [Input currency](#input-currency)
+  - [Sender](#sender)
+  - [Recipient](#recipient)
+    - [Payout type](#payout-type)
+    - [Requested amount and currency](#requested-amount-and-currency)
+    - [Payout details](#payout-details)
+  - [Metadata](#metadata)
+- [Transaction object](#transaction-object)
+  - [id](#id)
+  - [state](#state)
+  - [state_reason](#state-reason)
+  - [input_amount and input_currency](#input-amount-and-input-currency)
+  - [sender](#sender)
+  - [recipients](#recipients)
+- [Funding transactions](#funding-transactions)
+- [Checking the state of the transaction](#checking-the-state-of-the-transaction)
+- [Receiving error messages](#receiving-error-messages)
+- [Cancelling recipients and transactions](#cancelling-recipients-and-transactions)
+
+## Introduction
+
+Transactions are the main objects in the BitPesa API, so it's important to understand how to create and manage them. Transactions facilitate money movement from one Sender in a specific currency to one or multiple Recipients in another currency.
+
+The main flow of a successful transaction flow is the following:
+
+* Transaction is created linking the Sender to the Recipient(s) with the requested amounts.
+* Once the sender is KYC'd and approved the transaction can be funded.
+* Once the transaction is funded, we will initiate the payout to the recipient(s).
+* After the recipient (or all recipients) has received the money, the transaction is finished.
+
+Unfortunately not all transactions are successful. The main causes of issues are the following:
+
+* Transactions are cancelled automatically if they are not funded within the first hour of creation.
+* The recipient details might be wrong (for example the desitnation bank account number is invalid), or there are other issues blocking the successful payout. In these cases the transaction has to be cancelled.
+* Once the transaction is cancelled, we will refund the money to the sender. Dependent on how the transaction was funded this might be a manual process. Funds from account balances are always refunded automatically.
```

Comment:
Should this be a new bullet point? It seems like a continuation of the second bullet point, rather than being a cause of transactions failing itself.